include/nighthawk/common/nighthawk_sink_client.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include "envoy/common/pure.h"
+
+#include "external/envoy/source/common/common/statusor.h"
+#include "external/envoy/source/common/protobuf/protobuf.h"
+
+#include "api/client/options.pb.h"
+#include "api/client/service.grpc.pb.h"
+
+namespace Nighthawk {
+
+/**
+ * Interface of a gRPC sink service client.
+ */
+class NighthawkSinkClient {
+public:
+  virtual ~NighthawkSinkClient() = default;
+
+  /**
+   * @brief Store an execution response.
+   *
+   * @param nighthawk_sink_stub Used to open a channel to the sink service.
+   * @param store_execution_request Provide the message that the sink should store.
+   * @return absl::StatusOr<nighthawk::client::StoreExecutionResponse>
+   */
+  virtual absl::StatusOr<nighthawk::client::StoreExecutionResponse> StoreExecutionResponseStream(
+      nighthawk::client::NighthawkSink::StubInterface* nighthawk_sink_stub,
```

Comment:
Would a reference work better? (since it cannot be null).

Looks like the other method does accept a reference.

Reply:
Good catch; that was pretty inconsistent. Done in fe2eac357d600922489e611a8843548b544a16dd

Reply:
Err sorry f64a229b62bcaaa78b1316708fca4cf4324c3a1d